multi/imxrt-multi/uart.c

```diff
@@ -92,6 +101,22 @@ static const int uartPos[] = { UART1_POS, UART2_POS, UART3_POS, UART4_POS, UART5
 	UART7_POS, UART8_POS, UART9_POS, UART10_POS, UART11_POS, UART12_POS };
 
 
+static const uart_halfDuplexAction_t uartHalfDuplex[] = {
+	{ UART1_HALF_DUPLEX_GPIO },
+	{ UART2_HALF_DUPLEX_GPIO },
+	{ UART3_HALF_DUPLEX_GPIO },
+	{ UART4_HALF_DUPLEX_GPIO },
+	{ UART5_HALF_DUPLEX_GPIO },
+	{ UART6_HALF_DUPLEX_GPIO },
+	{ UART7_HALF_DUPLEX_GPIO },
+	{ UART8_HALF_DUPLEX_GPIO },
+	{ UART9_HALF_DUPLEX_GPIO },
```

Comment:
UART9-12 are available only on rt1176.

Reply:
UART9-12 are disabled by config.h when not on rt117x, so these defines don't hurt

Reply:
They consume memory.

Reply:
As do `uartConfig` and `uartPos`. Either all three should be rewritten or none at all, for consistency.

I think `uartConfig`, `uartPos`, `uartHalfDuplex`, and `uart_init():info` can all be merged into one `uartPreConfig` array, just like in i2c.c. @jmaksymowicz What do you think?

Reply:
I’m closing the topic since it doesn’t matter for the rt117x, but it might be worth adding a TODO note indicating that this could be optimized in the future.